README.md

```diff
@@ -1 +1,111 @@
-# orm-mapping-db-to-ruby-object
+#Mapping tables to objects 
+###Basics of reading from a database table that is mapped to a Ruby object
+
+
+##Objectives
+1. Build methods that read from a database table
+2. Understand how to build `Song.all` that returns all songs from the database
+3. Understand how to build `Song.find_by_name` method that return a song from the database by the song's name
+4. How to convert what the database gives you into a Ruby object
+
+##Overview
+Our Ruby program gets most interesting when we add data. To do this, we use a database. When we want our Ruby program to store things we send them off to a database. When we want to retrieve those things, we ask the database to send them back to our program. This works very well, but there is one small problem to overcome -- our Ruby program and the database don't speak the same language.
+
+Ruby understands objects. The database understands raw data.
+
+We don't store Ruby objects in the database, and we don't get Ruby objects back from the database. We store the raw data that describes a given Ruby object in a table row and we get back raw data that describes a ruby object when we select from that table. 
+
+When we query the database, it is up to us to write the code that takes that data and turns it back into an instance of whatever class. We, the programmers, will be the translators that translate the raw data that the database sends into Ruby objects that are instances of a particular class.
+
+##Example
+Let's use our song domain as an example. Imagine we have a Song class that is responsible for making songs. Every song will come with two attributes, a title and a length. We could make a bunch of new songs, but we want to look at all the songs we have that have already been created.
+
+Imagine we have a database with 1 million songs. We need to build three methods to access all of those songs and convert them to Ruby objects.
+
+##`.new_from_db`
+The first thing we need to do is convert what the database gives us into a Ruby object. We will use this method to create all the Ruby objects in our next two methods.
+
+The first thing to know is that the database (SQLite) in our case, will return an array of data for each row. For example, a row for Michael Jackson's "Thriller" (356 seconds long) that has a db id of 1 would look like this: `[1, "Thriller", 356]`.
+
+```ruby
+def self.new_from_db(row)
+  new_ruby_object = self.new  # self.new is the same as running Student.new
+  new_ruby_object.id = row[0]
+  new_ruby_object.name =  row[1]
+  new_ruby_object.length = row[2]
+end
+```
+
+This can also be achieved using the `tap` method. Tap yields the object to the block and then returns the original object. If you're comfortable using `tap`, go for it. If it's confusing, you can ignore it. 
```

Comment:
@jmburges Taking `tap` out. I thought it would be okay since theyve already done the Yield labs at this point, but I think including tap is unnecessary. Tap never gets you anything; it's just visually a bit cleaner.